File: pear.py
import inquirer
import pear_server as pear
import requests
import threading
import asyncio
import ipaddress
import sys
import os
import socket

def prompt():
    return input("> ")
"""
ip = requests.get("http://bot.whatismyipaddress.com")
ip = ip.text
print("Your IP addr",ip)
"""

serve = [
    inquirer.List('method',
                   message="Create server or connect to one?",
                   choices=['Create', 'Connect'])
]
answers = inquirer.prompt(serve)
if answers['method'] == 'Connect':
    # connect to server
    print("connecting")
else:
    serve_info = [
        inquirer.Text('host',
                       message="Choose a host to serve on",
                       default="ipv6"),
        inquirer.Text('port',
                       message="Choose a port to serve on, defaults to 4792",
                       default="4792")
    ]
    host_info = inquirer.prompt(serve_info)
    if host_info['host'] == "ipv6":
        host_info['host'] = ''
    ip = host_info['host']
    port = int(host_info['port'])

    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.bind((ip, port))
        s.listen()
        conn, addr = s.accept()
        with conn:
            print('Connected by', addr)
            async def input_handler():
                conn.sendall(sys.stdin.readline())
            async def read():
                while True:
                    data = conn.recv(1024)
                    if not data:
                        break
                    print(data.decode('utf8'))
            tasks = [
                asyncio.ensure_future(read()),
            ]
            loop = asyncio.get_event_loop()
            loop.add_reader(sys.stdin, input_handler)
            loop.run_until_complete(asyncio.wait(tasks))  
            loop.close()

File: pear.js
var net = require('net');
var inquirer = require('inquirer');
var readline = require('readline');

inquirer.prompt([
    {
        type: 'list',
        name: 'method',
        message: 'Create channel, or connect?',
        choices: [
            'Create',
            'Connect'
        ]
    }
]).then(answers => {
    if (answers.method == 'Create') {
        inquirer.prompt([
            {
                type: 'text',
                name: 'host',
                message: 'Choose the host',
                default: 'localhost'
            },
            {
                type: 'text',
                message: 'Choose the port',
                name: 'port',
                default: '9461'
            }
        ]).then(answers => {
            console.log("starting...")
            do_serve(parseInt(answers.port, 10), answers.host)
        })
    } else {
        inquirer.prompt([
            {
                type: 'text',
                name: 'host',
                message: 'Choose host to connect to'
            },
            {
                type: 'text',
                name: 'port',
                message: 'Choose port to connect to',
                default: '9461'
            }
        ]).then(answers => {
            do_connect(parseInt(answers.port, 10), answers.host)
        })
    }
})

function do_connect(port, host) {
    let socket = net.createConnection(port, host);
    socket.on("data", data => {
        console.log(data)
    })
    function inquire () {
        inquirer.prompt([
            {
                type: 'text',
                name: 'message',
                message: '>'
            }
        ]).then(answers => {
            socket.write(answers.message);
            inquire()
        })
    }
    inquire()
}

function do_serve(PORT, HOST) {
    net.createServer(function(sock) {
        // Add a 'data' event handler to this instance of socket
        sock.on('data', function(data) {
            
            console.log('DATA ' + sock.remoteAddress + ': ' + data);
            // Write the data back to the socket, the client will receive it as data from the server
            sock.write('You said "' + data + '"');
            
        });

        function inquire () {
            inquirer.prompt([
                {
                    type: 'text',
                    name: 'message',
                    message: ''
                }
            ]).then(answers => {
                sock.write(answers.message);
                inquire()
            })
        }
        inquire()
        
        // Add a 'close' event handler to this instance of socket
        sock.on('close', function(data) {
            console.log('CLOSED: ' + sock.remoteAddress +' '+ sock.remotePort);
        });
        
    }).listen(PORT, HOST);

    console.log('Server listening on ' + HOST +':'+ PORT);
}